common/lib/xmodule/xmodule/capa_module.py

```diff
@@ -301,6 +308,14 @@ def index_dictionary(self):
         Return dictionary prepared with module content and type for indexing.
         """
         xblock_body = super(ProblemBlock, self).index_dictionary()
+
+        # Make optioninput's options index friendly by replacing the actual tag with the values
+        capa_content = re.sub(
+            r'<optioninput options="\(([^"]+)\)".*?>\s*|\S*<\/optioninput>',
+            lambda matched: matched.group(1).replace("'", '').replace(",", ", ") if matched.group(1) else None,
```

Comment:
We need not strip out the commas and quotes since it will still find the words inside them.